File: flutter17_mobile/lib/models/television.dart
import 'package:json_annotation/json_annotation.dart';

part 'television.g.dart';

@JsonSerializable()
class Television {
  int? id;
  int? productId;

  // Display
  String? screenSize;
  String? screenResolution;
  String? screenType;
  bool? isSmartTV;
  int? refreshRate;
  bool? supportsHDR;

  // Audio
  int? speakerOutputPower;
  bool? supportsDolbyAtmos;

  // Connectivity
  int? hdmiInputs;
  int? usbPorts;
  bool? hasBluetooth;
  bool? hasWiFi;

  // Features
  String? operatingSystem;
  bool? supportsVoiceControl;
  bool? hasScreenMirroring;

  // Physical Characteristics
  double? weight;
  String? dimensions;
  String? standType;

  // Energy Efficiency
  String? energyRating;
  int? powerConsumption;

  Television({
    this.id,
    this.productId,
    this.screenSize,
    this.screenResolution,
    this.screenType,
    this.isSmartTV,
    this.refreshRate,
    this.supportsHDR,
    this.speakerOutputPower,
    this.supportsDolbyAtmos,
    this.hdmiInputs,
    this.usbPorts,
    this.hasBluetooth,
    this.hasWiFi,
    this.operatingSystem,
    this.supportsVoiceControl,
    this.hasScreenMirroring,
    this.weight,
    this.dimensions,
    this.standType,
    this.energyRating,
    this.powerConsumption,
  });

  factory Television.fromJson(Map<String, dynamic> json) => _$TelevisionFromJson(json);

  Map<String, dynamic> toJson() => _$TelevisionToJson(this);
}

File: angular17_web/src/app/components/see-more-details/spec-builders/television.ts
import { boolRow, row, SpecSection } from './shared';

export function televisionSections(p: any): SpecSection[] {
  return [
    {
      title: 'Display',
      rows: [
        row('Screen Size', p.screenSize),
        row('Resolution', p.screenResolution),
        row('Screen Type', p.screenType),
        boolRow('Smart TV', p.isSmartTV),
        row('Refresh Rate', p.refreshRate, ' Hz'),
        boolRow('HDR Support', p.supportsHDR),
      ],
    },
    {
      title: 'Audio',
      rows: [
        row('Speaker Output', p.speakerOutputPower, ' W'),
        boolRow('Dolby Atmos', p.supportsDolbyAtmos),
      ],
    },
    {
      title: 'Connectivity',
      rows: [
        row('HDMI Inputs', p.hdmiInputs),
        row('USB Ports', p.usbPorts),
        boolRow('Bluetooth', p.hasBluetooth),
        boolRow('WiFi', p.hasWiFi),
      ],
    },
    {
      title: 'Features',
      rows: [
        row('Operating System', p.operatingSystem),
        boolRow('Voice Control', p.supportsVoiceControl),
        boolRow('Screen Mirroring', p.hasScreenMirroring),
      ],
    },
    {
      title: 'Physical Characteristics',
      rows: [
        row('Weight', p.weight, ' kg'),
        row('Dimensions', p.dimensions),
        row('Stand Type', p.standType),
      ],
    },
    {
      title: 'Energy Efficiency',
      rows: [
        row('Energy Rating', p.energyRating),
        row('Power Consumption', p.powerConsumption, ' W'),
      ],
    },
  ];
}
